### PR TITLE
Giveth farms adapter

### PR DESCRIPTION
**Pull request by CarlosQ96**
Draft Pr for comments before sending the actual pr to defillama

### CODE REVIEW

File: src/adaptors/giveth/index.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    .div(2)
+    .div(tokenReserve);
+
+  const totalSupply = BigNumber(contractInfo.totalSupply);
+
+  const apr = totalSupply.isZero()
+    ? null
+    : BigNumber(contractInfo.rewardRate)
+        .div(totalSupply)
+        .times('31536000')
```
Comment (mohammadranjbarz) [inline src/adaptors/giveth/index.js:77]: @CarlosQ96 What is `31536000` ?
I think it's good to have a comment for these numbers

File: src/adaptors/giveth/index.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  return tvl;
+}
+
+const calculateUnipoolApy = async (entry) => {
+  const totalSupply = BigNumber(entry.totalSupply);
+  const rewardRate = BigNumber(entry.rewardRate);
+
+  const apr = totalSupply.isZero()
+    ? 0
+    : rewardRate.div(totalSupply).times('31536000').times('100');
```
Comment (mohammadranjbarz) [inline src/adaptors/giveth/index.js:108]: We used `31536000` twice, it's better to be in a field to can reuse it

**Comment by mohammadranjbarz** [comment]
@CarlosQ96 Thanks, functionality seems ok I ran giveth-adapter as you said, with this command

`node src/adaptors/test.js src/adaptors/giveth/index.js`
And I got this 
```
[
  {
    pool: '0xd93d3bdba18ebcb3317a57119ea44ed2cf41c2f2',
    chain: 'Xdai',
    project: 'giveth',
    symbol: 'GIV',
    tvlUsd: 7.884183054591371e+23,
    apy: 26.844725768615422
  },
  {
    pool: '0x4b9efae862a1755f7cecb021856d467e86976755',
    chain: 'Ethereum',
    project: 'giveth',
    symbol: 'GIV',
    tvlUsd: 1.2613493824046669e+23,
    apy: 55.93293825503789
  },
  {
    pool: '0xb7189a7ea38fa31210a79fe282aec5736ad5fa57',
    chain: 'Xdai',
    project: 'giveth',
    symbol: 'GIV-WXDAI',
    tvlUsd: 126628.38810434507,
    apy: 285.9805599194944
  },
  {
    pool: '0xbeba1666c62c65e58770376de332891b09461eeb',
    chain: 'Ethereum',
    project: 'giveth',
    symbol: 'DAI-GIV',
    tvlUsd: 104664.85359401311,
    apy: 353.3415840722072
  }
]
```
I compared that with https://giveth.io/givfarm farms 

Screen Shot 1401-05-05 at 16 45 11

Screen Shot 1401-05-05 at 16 45 58

And they seems ok , except I dont know why GIV/DAI apr is zero in givEconomy (maybe my network problem)

But I think it's better to @aminlatifi check it as well

File: src/adaptors/giveth/index.js (hunk 1 above)
Comment (CarlosQ96) [reply src/adaptors/giveth/index.js:77]: Its the seconds in a year! you are right ill add it as a constant.

**Comment by aminlatifi** [comment]
@mohammadranjbarz it's not zero at the moment
image

**Comment by aminlatifi** [comment]
@CarlosQ96 @mohammadranjbarz So can we merge it?

**Comment by mohammadranjbarz** [comment]
> @CarlosQ96 @mohammadranjbarz So can we merge it?

IMO yes, it seems good

**Comment by CarlosQ96** [comment]
Ill push changes and close this pr

**Comment by CarlosQ96** [comment]
Closing this older version

**Pull request closed without merging**